src/client/src/ui/common/regions/reducer.ts

```diff
@@ -1,34 +1,50 @@
 import { ACTION_TYPES as CONNECTION_ACTION_TYPES, DisconnectAction } from '../../../operations/connectionStatus'
 import { ACTION_TYPES, Region, RegionActions } from './actions'
 
-const changeRegionTearOffStatus = (state: RegionState, payload: Region, status: boolean) => ({
-  ...state,
-  [payload.id]: {
-    ...state[payload.id],
-    isTearedOff: status
-  }
-})
-
-export interface RegionState {
+export interface RegionsState {
   [regionId: string]: Region
 }
 
-const INITIAL_STATE: RegionState = {}
+const INITIAL_STATE: RegionsState = {}
 
-export const regionsReducer = (state: RegionState = {}, action: RegionActions | DisconnectAction): RegionState => {
+export const regionsReducer = (
+  state: RegionsState = INITIAL_STATE,
+  action: RegionActions | DisconnectAction
+): RegionsState => {
   switch (action.type) {
     case ACTION_TYPES.REGION_ADD:
-      return {
-        [action.payload.id]: action.payload,
-        ...state
-      }
     case ACTION_TYPES.REGION_ATTACH_WINDOW:
-      return changeRegionTearOffStatus(state, action.payload, false)
     case ACTION_TYPES.REGION_TEAROFF_WINDOW:
-      return changeRegionTearOffStatus(state, action.payload, true)
+      return { ...state, [action.payload.id]: regionReducer(state[action.payload.id], action) }
```

Comment:
Favour spreading code on to multiple lines if it makes it more readable.

Reply:
You do not need all these case statements here. Just send them down to the sub reducer.

Reply:
If we don't check for an appropriate action, the reducer would pass on actions that did not pertain to the regionReducer.